File: game.py
import pygame
import random
import time

# Initialize pygame
pygame.init()

# Colors
WHITE = (255, 255, 255)
BLACK = (0, 0, 0)
RED = (255, 0, 0)
GREEN = (0, 255, 0)

# Game settings
WIDTH, HEIGHT = 600, 400
BLOCK_SIZE = 20
FPS = 10

# Set up display
screen = pygame.display.set_mode((WIDTH, HEIGHT))
pygame.display.set_caption('Simple Snake Game')

clock = pygame.time.Clock()
font = pygame.font.SysFont('Arial', 25)

def draw_grid():
    for x in range(0, WIDTH, BLOCK_SIZE):
        for y in range(0, HEIGHT, BLOCK_SIZE):
            rect = pygame.Rect(x, y, BLOCK_SIZE, BLOCK_SIZE)
            pygame.draw.rect(screen, (50, 50, 50), rect, 1)

def game_loop():
    # Initial snake position (list of blocks)
    snake = [[WIDTH//2, HEIGHT//2]]
    snake_direction = [BLOCK_SIZE, 0]
    
    # Initial food position
    food = [
        random.randrange(0, WIDTH - BLOCK_SIZE, BLOCK_SIZE),
        random.randrange(0, HEIGHT - BLOCK_SIZE, BLOCK_SIZE)
    ]
    
    score = 0
    game_over = False
    
    while True:
        for event in pygame.event.get():
            if event.type == pygame.QUIT:
                pygame.quit()
                return
            
            if event.type == pygame.KEYDOWN:
                if event.key == pygame.K_UP and snake_direction[1] != BLOCK_SIZE:
                    snake_direction = [0, -BLOCK_SIZE]
                if event.key == pygame.K_DOWN and snake_direction[1] != -BLOCK_SIZE:
                    snake_direction = [0, BLOCK_SIZE]
                if event.key == pygame.K_LEFT and snake_direction[0] != BLOCK_SIZE:
                    snake_direction = [-BLOCK_SIZE, 0]
                if event.key == pygame.K_RIGHT and snake_direction[0] != -BLOCK_SIZE:
                    snake_direction = [BLOCK_SIZE, 0]
                if event.key == pygame.K_r and game_over:
                    return game_loop()
        
        if game_over:
            continue
        
        # Move snake
        new_head = [snake[0][0] + snake_direction[0], snake[0][1] + snake_direction[1]]
        snake.insert(0, new_head)
        
        # Check if snake ate food
        if snake[0] == food:
            score += 1
            food = [
                random.randrange(0, WIDTH - BLOCK_SIZE, BLOCK_SIZE),
                random.randrange(0, HEIGHT - BLOCK_SIZE, BLOCK_SIZE)
            ]
            # Make sure food doesn't spawn on snake
            while food in snake:
                food = [
                    random.randrange(0, WIDTH - BLOCK_SIZE, BLOCK_SIZE),
                    random.randrange(0, HEIGHT - BLOCK_SIZE, BLOCK_SIZE)
                ]
        else:
            snake.pop()
        
        # Check collisions
        if (snake[0][0] < 0 or snake[0][0] >= WIDTH or
            snake[0][1] < 0 or snake[0][1] >= HEIGHT or
            snake[0] in snake[1:]):
            game_over = True
        
        # Draw everything
        screen.fill(BLACK)
        draw_grid()
        
        # Draw snake
        for block in snake:
            pygame.draw.rect(screen, GREEN, (block[0], block[1], BLOCK_SIZE, BLOCK_SIZE))
        
        # Draw food
        pygame.draw.rect(screen, RED, (food[0], food[1], BLOCK_SIZE, BLOCK_SIZE))
        
        # Draw score
        score_text = font.render(f'Score: {score}', True, WHITE)
        screen.blit(score_text, (10, 10))
        
        if game_over:
            game_over_text = font.render('Game Over! Press R to restart', True, WHITE)
            screen.blit(game_over_text, (WIDTH//2 - 150, HEIGHT//2))
        
        pygame.display.update()
        clock.tick(FPS)

# Start the game
game_loop()

File: game.js
// Game constants
const GRID_SIZE = 20;
const CANVAS_SIZE = 400;
const FPS = 10;

// Game state
let snake = [{x: 200, y: 200}];
let food = generateFood();
let dx = GRID_SIZE;
let dy = 0;
let score = 0;
let gameOver = false;
let gameInterval;

// DOM elements
const canvas = document.getElementById('gameCanvas');
const ctx = canvas.getContext('2d');
const scoreDisplay = document.getElementById('score');
const restartButton = document.getElementById('restart');

// Initialize game
function init() {
    document.addEventListener('keydown', changeDirection);
    restartButton.addEventListener('click', resetGame);
    gameInterval = setInterval(gameLoop, 1000 / FPS);
}

// Main game loop
function gameLoop() {
    if (gameOver) return;
    
    moveSnake();
    checkCollision();
    drawGame();
}

// Move the snake
function moveSnake() {
    const head = {x: snake[0].x + dx, y: snake[0].y + dy};
    snake.unshift(head);
    
    // Check if snake ate food
    if (head.x === food.x && head.y === food.y) {
        score++;
        scoreDisplay.textContent = `Score: ${score}`;
        food = generateFood();
    } else {
        snake.pop();
    }
}

// Generate random food position
function generateFood() {
    const position = {
        x: Math.floor(Math.random() * (CANVAS_SIZE / GRID_SIZE)) * GRID_SIZE,
        y: Math.floor(Math.random() * (CANVAS_SIZE / GRID_SIZE)) * GRID_SIZE
    };
    
    // Make sure food doesn't spawn on snake
    while (snake.some(segment => segment.x === position.x && segment.y === position.y)) {
        position.x = Math.floor(Math.random() * (CANVAS_SIZE / GRID_SIZE)) * GRID_SIZE;
        position.y = Math.floor(Math.random() * (CANVAS_SIZE / GRID_SIZE)) * GRID_SIZE;
    }
    
    return position;
}

// Check for collisions
function checkCollision() {
    const head = snake[0];
    
    // Wall collision
    if (head.x < 0 || head.x >= CANVAS_SIZE || head.y < 0 || head.y >= CANVAS_SIZE) {
        endGame();
    }
    
    // Self collision
    for (let i = 1; i < snake.length; i++) {
        if (head.x === snake[i].x && head.y === snake[i].y) {
            endGame();
        }
    }
}

// Handle keyboard input
function changeDirection(event) {
    const key = event.keyCode;
    
    // Prevent reverse direction
    if (key === 37 && dx === 0) { // Left
        dx = -GRID_SIZE;
        dy = 0;
    } else if (key === 38 && dy === 0) { // Up
        dx = 0;
        dy = -GRID_SIZE;
    } else if (key === 39 && dx === 0) { // Right
        dx = GRID_SIZE;
        dy = 0;
    } else if (key === 40 && dy === 0) { // Down
        dx = 0;
        dy = GRID_SIZE;
    }
}

// Draw game elements
function drawGame() {
    // Clear canvas
    ctx.fillStyle = 'black';
    ctx.fillRect(0, 0, CANVAS_SIZE, CANVAS_SIZE);
    
    // Draw snake
    snake.forEach((segment, index) => {
        ctx.fillStyle = index === 0 ? '#4CAF50' : '#8BC34A'; // Head darker than body
        ctx.fillRect(segment.x, segment.y, GRID_SIZE, GRID_SIZE);
    });
    
    // Draw food
    ctx.fillStyle = '#FF5722';
    ctx.fillRect(food.x, food.y, GRID_SIZE, GRID_SIZE);
}

// End game
function endGame() {
    gameOver = true;
    clearInterval(gameInterval);
    alert(`Game Over! Your score: ${score}`);
}

// Reset game
function resetGame() {
    snake = [{x: 200, y: 200}];
    food = generateFood();
    dx = GRID_SIZE;
    dy = 0;
    score = 0;
    gameOver = false;
    scoreDisplay.textContent = `Score: ${score}`;
    clearInterval(gameInterval);
    gameInterval = setInterval(gameLoop, 1000 / FPS);
}

// Start the game
init();